[macOS Debug wk2] ASSERTION FAILED: !m_messageReceiverMapCount in webrtc tests

youennf opened the pull request:
#### bbf8a24839a069de0713ce35b0de6321e8a3bfd9
<pre>
[macOS Debug wk2] ASSERTION FAILED: !m_messageReceiverMapCount in webrtc tests
<a href="https://rdar.apple.com/140302668">rdar://140302668</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=283445">https://bugs.webkit.org/show_bug.cgi?id=283445</a>

Reviewed by Chris Dumez.

RemoteRemoteCommandListener is already fast malloced so does not need to be TZone malloced.
This seems to confuse our weak pointer implementation, which makes RemoteRemoteCommandListener no longer always receives GPUProcessConnection close notifications.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebKit/WebProcess/GPU/media/RemoteRemoteCommandListener.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteRemoteCommandListener.h:

Canonical link: <a href="https://commits.webkit.org/286967@main">https://commits.webkit.org/286967@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e70e47dc3eac56aca6b0c4a0d7a81626d51af69

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77695 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56728 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/30617 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/82369 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28973 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65884 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5028 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60898 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18836 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80761 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50843 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66664 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41189 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48220 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/24170 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/27304 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69336 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/24518 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/83699 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5075 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/3434 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69099 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/5233 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66646 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68350 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17081 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12360 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/10458 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5022 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/7817 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5035 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/8473 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6800 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
I think this may be the bug that @kmiller68 is working on (TZone and ThreadSafeWeakHashSet not playing well together).

youennf committed:
[macOS Debug wk2] ASSERTION FAILED: !m_messageReceiverMapCount in webrtc tests
rdar://140302668
https://bugs.webkit.org/show_bug.cgi?id=283445

Reviewed by Chris Dumez.

RemoteRemoteCommandListener is already fast malloced so does not need to be TZone malloced.
This seems to confuse our weak pointer implementation, which makes RemoteRemoteCommandListener no longer always receives GPUProcessConnection close notifications.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebKit/WebProcess/GPU/media/RemoteRemoteCommandListener.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteRemoteCommandListener.h:

Canonical link: https://commits.webkit.org/286967@main
